### PR TITLE
Add errors

### PR DESCRIPTION
**Pull request by chandr-andr**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -12,6 +12,8 @@ license = "LICENSE"
 python = "^3.7"
 typing-extensions = ">=3.10.0.0"
 pydantic = "^1.6.2"
+redis = "^4.3.4"
```
Comment (s3rius) [inline pyproject.toml:15]: Эт зачем это тут?

Comment (chandr-andr) [reply pyproject.toml:15]: А как ошибки редиски ловить?

Comment (s3rius) [inline pyproject.toml:15]: Please delete this. 

We don't want any extra dependency in core repo.

File: pyproject.toml [hunk 2]
```diff
@@ -40,6 +42,10 @@ implicit_reexport = true
 allow_untyped_decorators = true
 warn_return_any = false
 
+[[tool.mypy.overrides]]
```
Comment (s3rius) [inline pyproject.toml:45]: image

File: taskiq/abc/broker.py [hunk 3]
```diff
@@ -121,13 +123,18 @@ async def kiq(
...
         :returns: taskiq task.
         """
         logger.debug(
             f"Kicking {self.task_name} with args={args} and kwargs={kwargs}.",
         )
         message = self._prepare_message(*args, **kwargs)
-        await self.broker.kick(message)
+        try:
+            await self.broker.kick(message)
+        except AMQPConnectionError as exc:
```
Comment (s3rius) [inline taskiq/abc/broker.py:136]: Please catch `Exception` not specific exceptions. 

What if someone will implement kafka broker? We don't want to have kafka in dependencies in taskiq and moreover we don't want to add extra except clause.

**Review by s3rius — CHANGES_REQUESTED** [review]
You have to be more generic. Since taskiq gives people ability to easily add support for their technologies. We don't want to catch specific exceptions. We want to catch all of them.

Also please remove redundant dependencies. Core repo must be as small as possible.

File: pyproject.toml (hunk 1 above)
Comment (chandr-andr) [reply pyproject.toml]: Deleted

File: pyproject.toml (hunk 2 above)
Comment (chandr-andr) [reply pyproject.toml]: Deleted it 1 minute ago))

File: taskiq/abc/broker.py (hunk 3 above)
Comment (chandr-andr) [reply taskiq/abc/broker.py]: Changed specific exceptions to `Exception`

File: taskiq/cli/async_task_runner.py [hunk 4]
```diff
@@ -277,3 +280,4 @@ async def async_listen_messages(  # noqa: C901, WPS210, WPS213
             await broker.result_backend.set_result(message.task_id, result)
         except Exception as exc:
             logger.exception(exc)
+            raise ResultSetError() from exc
```
Comment (s3rius) [inline taskiq/cli/async_task_runner.py:283]: I think it would be better not to raise exception here. Since it will kill the worker's process.

**Review by s3rius — CHANGES_REQUESTED** [review]
A little change to make workers more stable.

File: taskiq/cli/async_task_runner.py (hunk 4 above)
Comment (chandr-andr) [reply taskiq/cli/async_task_runner.py:283]: Okay, but maybe we can change logger message to more understandable? I think that people need to explicitly understand that the broker is broken and cannot accept the message

**Comment by chandr-andr** [comment]
> A little change to make workers more stable.

Done, please have a look

File: taskiq/cli/async_task_runner.py [hunk 5]
```diff
@@ -275,4 +275,9 @@ async def async_listen_messages(  # noqa: C901, WPS210, WPS213
         try:
             await broker.result_backend.set_result(message.task_id, result)
         except Exception as exc:
-            logger.exception(exc)
+            logger.exception(
```
Comment (s3rius) [inline taskiq/cli/async_task_runner.py:278]: You better do this:

```python
logger.error(
    "Can't set result in %s result backend: \n %s.",
    broker.result_backend,
    exc,
    exc_info=True
)
```

**Review by s3rius — APPROVED** [review]
LGTM.

**Pull request merged**